TrafficCapture/trafficReplayer/src/test/java/org/opensearch/migrations/replay/ResultsToLogsConsumerTest.java

```diff
@@ -20,52 +28,55 @@
 import org.opensearch.migrations.testutils.WrapWithNettyLeakDetection;
 import org.opensearch.migrations.tracing.InstrumentationTest;
 import org.opensearch.migrations.tracing.TestContext;
-
-import java.io.IOException;
-import java.io.InputStream;
-import java.nio.charset.StandardCharsets;
-import java.time.Duration;
-import java.time.Instant;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.stream.Collectors;
+import org.slf4j.LoggerFactory;
 
 @Slf4j
 @WrapWithNettyLeakDetection(repetitions = 4)
 class ResultsToLogsConsumerTest extends InstrumentationTest {
     private static final String NODE_ID = "n";
     private static final ObjectMapper mapper = new ObjectMapper();
     public static final String TEST_EXCEPTION_MESSAGE = "TEST_EXCEPTION";
+    private static final Random RANDOM = new Random();
 
     @Override
     protected TestContext makeInstrumentationContext() {
         return TestContext.withTracking(false, true);
     }
 
     private static class CloseableLogSetup implements AutoCloseable {
-        List<String> logEvents = new ArrayList<>();
+        List<String> logEvents = Collections.synchronizedList(new ArrayList<>());
         AbstractAppender testAppender;
+        org.slf4j.Logger testLogger;
+        org.apache.logging.log4j.core.Logger internalLogger;
+
+        final String instanceName;
+
         public CloseableLogSetup() {
-            testAppender = new AbstractAppender(ResultsToLogsConsumer.OUTPUT_TUPLE_JSON_LOGGER,
+            instanceName = this.getClass().getName() + "." + RANDOM.nextLong();
+
+            testAppender = new AbstractAppender(instanceName,
                     null, null, false, null) {
                 @Override
                 public void append(LogEvent event) {
                     logEvents.add(event.getMessage().getFormattedMessage());
                 }
             };
-            var tupleLogger = (Logger) LogManager.getLogger(ResultsToLogsConsumer.OUTPUT_TUPLE_JSON_LOGGER);
-            tupleLogger.setLevel(Level.ALL);
+
             testAppender.start();
-            tupleLogger.setAdditive(false);
-            tupleLogger.addAppender(testAppender);
-            var loggerCtx = ((LoggerContext) LogManager.getContext(false));
+
+            internalLogger = (org.apache.logging.log4j.core.Logger) LogManager.getLogger(instanceName);
+            testLogger = LoggerFactory.getLogger(instanceName);
```

Comment:
When running tests in parallel, using the default static logger instance caused collisions between tests. Here we are instantiating a test-specific logger to override the internal logger